perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -3183,12 +3185,15 @@ private void processMemberAfterRemovingLastSponsor(PerunSession sess, Member spo
 
 			// we must pass member back for the purpose of validation
 			status.put(MEMBER, member);
-
+			if (groups != null) {
+				perunBl.getGroupsManagerBl().addMember(sess, groups, member);
```

Comment:
I would move this to the end to own try/catch, because it shouldn't be logged as error during sponsored user creation. 

Also when you iterate over the groups here, then we know which groups failed and which were ok and can report this back to GUI. Right now you also stop adding member to group on first error, which might later require more work for admin to get to expected state (find out and add members to all expected groups manually).

Reply:
I have implemented these changes. In the returned results, for each member, there is a map of failed group assignments, where the key is the group id and the value is the error message. This can be used in GUI to show information about the assignments that failed.